module/module.go

```diff
@@ -776,6 +776,7 @@ func (m *Module) RemoveResource(ctx context.Context, req *pb.RemoveResourceReque
 
 	delete(m.streamSourceByName, res.Name())
 	delete(m.activeResourceStreams, res.Name())
+	delete(m.resLoggers, res.Name())
```

Comment:
Is it guaranteed that all requests for a resource are drained before `RemoveResource` can be processed?

Reply:
That's a good question; I believe the answer is no. I just tried `RemoveResource`ing a Golang modular component in the middle of its `DoCommand` invocation. We certainly don't have any logic now that waits for that `DoCommand` to finish before completing `RemoveResource`.

Are you asking that question because you're concerned that `delete`ing the logger from the `resLoggers` map could cause the logger to be garbage collected while a request is still using it?